zng/ztests/enum.yaml

```diff
@@ -0,0 +1,13 @@
+zql: "put s=e:string, v=e+1"
```

Comment:
So it looks like for an enum-valued `e`, to get the name you have to coerce to string, whereas `e` itself is the typed value ? This would be quite different both from  c-like enums and from sum types. 
I'm thinking it might be best to have c-like enums with a name and an (implicit) integer index, where only the name is visible in ZQL.

Reply:
C enums aren't necessarily a positional and can take initializers.  I was trying to capture the enum type as a place to interface with languages that have enums in them, which is I why I made them more general.  (We talked about rust enum previously, but I think that can be captured as a zng union). 

Here, a goal is to be able to put enums in and get enums out, for most style of enums out there.  If you google around for enum relating to spark, parquet, avro, mongo, OLAP, etc they do come up but perhaps the use cases are niche.

The avro definitioin of enum follows your proposal.  Parquet and mongo do not have enums.

I made cast work to get the name so you could do introspection, e.g., so you could do 
```
count() by e:string
```
and get something reasonable.  Normally the name of the enum is the useful thing and the value is useful for the code.  So if we just assume int, you'd still want to get at the name. 
I do realize e:string is a little odd if the value is also string.  Maybe we could define a function to give the name?  As I mentioned on today's call, I don't see this as something we would use in the near term but this fell in the scope of converging to a final zng spec.

Reply:
MySql enums follow your proposal as well.  SQL Server and Snowflake, as examples, don't seem to support enum.

Reply:
> So if we just assume int, you'd still want to get at the name.

I think this is the heart of the matter: if we just assume int, do you ever want anything other than the name? I'm assuming not, which is why i'm thinking `e` evaluates to the name (or label).


> We talked about rust enum previously, but I think that can be captured as a zng union

(I agree that zng unions are in spirit comparable to rust enums aka sum types, but they're not as expressive. I'm assuming that's ok...  if we want to have a data type system that cleanly maps onto an algebraic type system, that's a whole new can of worms).

Reply:
Agreed. Let's get this version of eunm in the commit history in this general way then we can change or delete.

Reply:
👍